### DISCUSSION TITLE
Feature Request: Add last business day option to time range filter

### DISCUSSION THREAD

**Discussion by NikolayZakharevich** [apache/superset#22427] [Ideas]
Hi all, I would like to reopen feature request #13261 

**Is your feature request related to a problem? Please describe.**
I am working on a dataset that does not have data for weekends. Currently, the time range filters supported are based on the calendar irrespective of the presence of data. If I search on a Monday using the last day filter then no results were returned, as expected. 
The difference from #13261 is that I want to get not only the last business day for Monday but also for all the rest days of the week.

**Describe the solution you'd like**
It would be great if there was an option to filter based on the availability of data or the last business day. The option should return:

```
on Monday: Friday <= col <= Friday
on Tuesday: Monday <= col <= Monday
on Wednesday: Tuesday <= col <= Tuesday
on Thursday: Wednesday <= col <= Wednesday
on Friday: Thursday <= col <= Thursday
on Saturday: Friday <= col <= Friday
on Sunday: Friday <= col <= Friday
```
This behaviour is originally described in https://github.com/apache/superset/issues/13261#issuecomment-1321736535

**Describe alternatives you've considered**
I tried to use Advanced date functions as suggested by @ktmud in https://github.com/apache/superset/issues/13261#issuecomment-782975758
```
dateadd(datetrunc(datetime('today'), week), 5, day)
```
but it works only on Mondays

**Additional context** 
![Image of Time Range](https://camo.githubusercontent.com/8946489e43bfd2cc8da067bcdadd8ab935deca6b7b8874336cd0a7a9b416112a/68747470733a2f2f692e696d6775722e636f6d2f36366e76524f712e706e67)

**Comment by rusackas** [comment]
CC @kasiazjc @lauderbaugh for future time picker design mods.

**Comment by TerraNibble** [comment]
@NikolayZakharevich did you find a solution to this?

**Comment by bt-88** [comment]
You can use a custom SQL expression in your time filter. It also works in the 'Advanced' filter.

For my _MariaDb_ queries, I use an expression like this.

```sql
measured_at > IF(DAYOFWEEK(CURDATE()) = 2, DATE_SUB(CURDATE(), INTERVAL 3 DAY),IF(DAYOFWEEK(CURDATE()) = 1, DATE_SUB(CURDATE(), INTERVAL 2 DAY),DATE_SUB(CURDATE(), INTERVAL 1 DAY)))
```
